stratum/hal/lib/common/yang_parse_tree_paths.cc

```diff
@@ -926,18 +926,47 @@ void SetUpInterfacesInterfaceStateLastChange(TreeNode* node) {
 
 ////////////////////////////////////////////////////////////////////////////////
 // /interfaces/interface[name=<name>]/state/ifindex
-void SetUpInterfacesInterfaceStateIfindex(uint32 port_id, TreeNode* node) {
+void SetUpInterfacesInterfaceStateIfindex(uint32 node_id, uint32 port_id,
+                                          TreeNode* node, YangParseTree* tree) {
+  // Returns the port ID for the interface to be used by P4Runtime.
+  // If Stratum performs port translation (e.g. for the bcm target), we return
+  // the port_id provided by the ChassisConfig (also called SDN port ID).
+  // If Stratum does not perform port translation (e.g. for the bf target), we
+  // return the SDK port number.
+  auto on_poll_functor = [tree, port_id, node_id](const GnmiEvent& event,
+                                                  const ::gnmi::Path& path,
+                                                  GnmiSubscribeStream* stream) {
+    // Build data request to see if the switch overrides the SDN port ID.
+    DataRequest req;
+    auto* port_req = req.add_requests()->mutable_sdn_port_id_override();
+    port_req->set_node_id(node_id);
+    port_req->set_port_id(port_id);
+
+    // Build the data response function to extract the overridden port ID, if
+    // the switch implementation chooses to do so.
+    uint32 sdn_port = port_id;
+    DataResponseWriter writer([&sdn_port](const DataResponse& resp) {
+      if (resp.has_sdn_port_id_override()) {
+        sdn_port = resp.sdn_port_id_override().port_id();
+        return true;
+      }
+      return false;
+    });
+
+    // Query the switch for the overridden port ID.
+    std::vector<::util::Status> details;
+    RETURN_IF_ERROR(tree->GetSwitchInterface()->RetrieveValue(
```

Comment:
Depending on if/how we intend to support P4RT translations, we might have another mechanism for getting this data into the functor in the future.